python/pyspark/streaming/dstream.py

```diff
@@ -51,122 +76,165 @@ class DStream(object):
      - A function that is used to generate an RDD after each time interval
     """
 
-    def __init__(self, jdstream, ssc, jrdd_deserializer):
+    def __init__(
+        self,
+        jdstream: JavaObject,
+        ssc: "StreamingContext",
+        jrdd_deserializer: "Serializer",
```

Comment:
Same as https://github.com/apache/spark/pull/34293/files#r787298665.

Would you address the all such cases below??

Reply:
For starters, this should be re-synced with master, as covariance has some impact on annotations on implementation (I took over SPARK-37154 from Byron to deal with that on `pyspark.rdd`).